lib/publiccloud/gcp_client.pm

```diff
@@ -71,7 +72,27 @@ sub get_next_vault_role {
 sub create_credentials_file {
     my ($self) = @_;
     my $credentials_file;
+    if (get_var('PUBLIC_CLOUD_CREDENTIALS_URL')) {
+        my $json = get_credentials();
+        # Need to remove quotes, convert \n to \\n and / to \/
+        assert_script_run(q{export key=$(cat /root/credentials.json | jq '.private_key_data' | sed 's/\\"//g' | sed 's/\\\n/\\\\\\\n/g' | sed 's/\\//\\\\\//g')});
```

Comment:
The idea is to not export secrets on the log. 

Exists the function define_secret_variable in publiccloud/utils
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14797

Also, I am not sure if executing this is going to be printed on the logs

Reply:
It doesn't print anything on the logs, that's why I use those `sed` pipes instead of 
```
my $output = script_output('cat /root/credentials.json');
# do some perl magic to extract the values
```
which would put everything in the logs.

Reply:
The problem of `define_secret_variable` is that it expects a variable name and a variable secret. 
What I need to do is to extract the secret from some json file...